packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -388,7 +388,26 @@ abstract class Route<T> {
   @mustCallSuper
   void changedExternalState() { }
 
-  /// Discards any resources used by the object.
+  bool _holdingGracefulDisposal = false;
+  /// Set whether [gracefullyDispose] should call [dispose] at a later time.
+  ///
+  /// Calling [markHoldingGracefulDisposal] with false will invoke [dispose] if
+  /// [gracefullyDispose] has been called while [markHoldingGracefulDisposal] is
+  /// set to true. See [gracefullyDispose] for detailed usage.
+  @protected
+  void markHoldingGracefulDisposal(bool value) {
```

Comment:
What if two entities want this to be graceful? Once the first one has given up on their request for gracefullness, the second one would be out of luck?

Reply:
(Took me a while to understand but the better way to phrase it is probably "what if two entities want to hold graceful disposal")

You're right. I should probably use a counter instead.